src/System.Buffers/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace System.Buffers
+{
+    [EventSource(
+        Guid = "20b30044-b729-457e-8dda-8b41b5dd02e6", 
+        Name = "System.Buffers.DefaultArrayPoolEvents",
+        LocalizationResources = "FxResources.System.Buffers.SR")]
+    internal sealed class ArrayPoolEventSource : EventSource
+    {
+        internal readonly static ArrayPoolEventSource Log = new ArrayPoolEventSource();
```

Comment:
Since the ArrayPool design allows folks to construct their own ArrayPool instance, should an ArrayPool itself have an internal ID that's traced out with every operation on that pool?  That way the events for each pool instance could be distinguished, a consumer would be able to tell how many different pools were being used, what kind of activity there was on each, etc.?